app/services/WhatYouOweService.scala

```diff
@@ -56,33 +57,40 @@ class WhatYouOweService @Inject()(val financialDetailsService: FinancialDetailsS
     }.flatten
   }
 
-  def getWhatYouOweChargesList(unpaidCharges: List[FinancialDetailsResponseModel],
+  def getWhatYouOweChargesList(unpaidCharges: Option[FinancialDetailsResponseModel],
                                isReviewAndReconciledEnabled: Boolean,
                                isFilterCodedOutPoasEnabled: Boolean)
                               (implicit headerCarrier: HeaderCarrier, mtdUser: MtdItUser[_]): Future[WhatYouOweChargesList] = {
 
     unpaidCharges match {
-      case financialDetails: List[FinancialDetailsResponseModel] if financialDetails.exists(_.isInstanceOf[FinancialDetailsErrorModel]) =>
-        throw new Exception("Error response while getting Unpaid financial details")
-      case financialDetails =>
-        val financialDetailsModelList = financialDetails.asInstanceOf[List[FinancialDetailsModel]]
-        val balanceDetails = financialDetailsModelList.headOption
-          .map(_.balanceDetails).getOrElse(BalanceDetails(0.00, 0.00, 0.00, None, None, None, None, None))
+      case Some(fds: FinancialDetailsModel) =>
+        val balanceDetails = fds.balanceDetails
         val codedOutChargeItem = {
-          financialDetailsModelList.flatMap(_.toChargeItem())
+          fds.toChargeItem()
             .filter(_.subTransactionType.contains(Accepted))
             .find(_.taxYear.endYear == (dateService.getCurrentTaxYearEnd - 1))
         }
 
         val whatYouOweChargesList = WhatYouOweChargesList(
           balanceDetails = balanceDetails,
-          chargesList = getFilteredChargesList(financialDetailsModelList, isReviewAndReconciledEnabled, isFilterCodedOutPoasEnabled),
+          chargesList = getFilteredChargesList(fds, isReviewAndReconciledEnabled, isFilterCodedOutPoasEnabled),
           codedOutDocumentDetail = codedOutChargeItem)
 
         callOutstandingCharges(mtdUser.saUtr, mtdUser.incomeSources.yearOfMigration, dateService.getCurrentTaxYearEnd).map {
           case Some(outstandingChargesModel) => whatYouOweChargesList.copy(outstandingChargesModel = Some(outstandingChargesModel))
           case _ => whatYouOweChargesList
         }
+      case Some(errorModel: FinancialDetailsErrorModel) =>
+        Logger("application").error(s"error model in unpaid charges - code: ${errorModel.code}, message: ${errorModel.message}")
+        throw new Exception("Error response while getting Unpaid financial details")
+      case None =>
+        val whatYouOweChargesList = WhatYouOweChargesList(BalanceDetails(0.00, 0.00, 0.00, None, None, None, None, None))
+        callOutstandingCharges(mtdUser.saUtr, mtdUser.incomeSources.yearOfMigration, dateService.getCurrentTaxYearEnd).map {
```

Comment:
passing optional UTR and then used it like: -> saUtr.get, maybe not a good idea, same for passing year of migration
yearOfMigration.get... 
callOutstandingCharges method need to be re-design, you can create a tech debt for this or fix in scope of this change